registry/image.go

```diff
@@ -0,0 +1,128 @@
+package registry
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+const (
+	dockerHubHost    = "index.docker.io"
+	dockerHubLibrary = "library"
+)
+
+// An image is a fully qualified name that refers to a particular image.
+// It is in the format: host[:port]/orgname/reponame[:tag]
+// Here, we refer to the "name" == orgname/reponame
+
+// Image can't really be a primitive string only, because we need to also
+// record information about it's creation time. (maybe more in the future)
+type Image interface {
+	WithTag(tag string) Image
+	WithCreatedAt(t *time.Time) Image
+	Components() (host, org, repo, tag string)
+	Host() string
+	Org() string
+	Repo() string
+	Tag() string
+	FQN() string
+	HostOrgRepo() string
+	OrgRepo() string
+	CreatedAt() *time.Time
+	Clone() Image
+}
+
+type image struct {
+	host, org, repo, tag string
+	createdAt            *time.Time `json:",omitempty"`
+}
```

Comment:
Interfaces are better for testing and easier to refactor.

An image belongs with a registry in the domain. It doesn't belong in the root. And also, you'd get circular dependencies (flux.Service->registry.x->flux.Image)